### DISCUSSION TITLE
Found bug with woocommerce module Elementor Pro

### DISCUSSION THREAD

**Discussion by graphem** [elementor/elementor#19165] [Developers Community]
Line 1619 of elementor-pro/modules/woocommerce/widgets/purchase-summary.php 

Need to include that variable to the query: 'type' => 'shop_order', so it look like this:

$latest_order = wc_get_orders( [
    'limit' => 1,
    'type' => 'shop_order',
    'orderby'  => 'date',
    'order'    => 'DESC',
    'return'   => 'ids',
] );

Otherwise it crash when the store has refund, as it is getting order refund in the query and different object:

PHP Fatal error:  Uncaught Error: Call to undefined method Automattic\WooCommerce\Admin\Overrides\OrderRefund::get_order_key() in /wp-content/plugins/elementor-pro/modules/woocommerce/widgets/purchase-summary.php:1636

**Comment by ppwebox** [comment]
Good catch and fix, thank you graphem! 

It's close to the end of 2024 and this nasty bug has still not been fixed! Bad, Elementor! We had a refund issued via Woo today, then wanted to edit the thank you page in Elementor Pro, which then caused the site to crash. 

Log entry: 
Critical Uncaught Error: Call to undefined method Automattic\WooCommerce\Admin\Overrides\OrderRefund::get_order_key() in /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor-pro/modules/woocommerce/widgets/purchase-summary.php:1726

Additional context
{
    "error": {
        "type": 1,
        "file": "/home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor-pro/modules/woocommerce/widgets/purchase-summary.php",
        "line": 1726
    },
    "backtrace": [
        "",
        "#0 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor-pro/modules/woocommerce/widgets/purchase-summary.php(1650): ElementorPro\Modules\Woocommerce\Widgets\Purchase_Summary->set_preview_order()",
        "#1 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/includes/base/controls-stack.php(2325): ElementorPro\Modules\Woocommerce\Widgets\Purchase_Summary->render()",
        "#2 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/includes/base/widget-base.php(635): Elementor\Controls_Stack->render_by_mode()",
        "#3 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/includes/base/widget-base.php(759): Elementor\Widget_Base->render_content()",
        "#4 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/includes/base/element-base.php(618): Elementor\Widget_Base->get_raw_data()",
        "#5 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/includes/base/element-base.php(618): Elementor\Element_Base->get_raw_data()",
        "#6 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/core/base/document.php(1085): Elementor\Element_Base->get_raw_data()",
        "#7 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/core/base/document.php(738): Elementor\Core\Base\Document->get_elements_raw_data()",
        "#8 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/includes/base/controls-stack.php(1070): Elementor\Core\Base\Document->get_initial_config()",
        "#9 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/core/editor/loader/common/editor-common-scripts-settings.php(38): Elementor\Controls_Stack->get_config()",
        "#10 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/core/editor/loader/v1/editor-v1-loader.php(66): Elementor\Core\Editor\Loader\Common\Editor_Common_Scripts_Settings::get()",
        "#11 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/core/editor/editor.php(364): Elementor\Core\Editor\Loader\V1\Editor_V1_Loader->enqueue_scripts()",
        "#12 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/class-wp-hook.php(324): Elementor\Core\Editor\Editor->enqueue_scripts()",
        "#13 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()",
        "#14 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/plugin.php(517): WP_Hook->do_action()",
        "#15 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/script-loader.php(2263): do_action()",
        "#16 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/class-wp-hook.php(324): wp_enqueue_scripts()",
        "#17 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()",
        "#18 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/plugin.php(517): WP_Hook->do_action()",
        "#19 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/general-template.php(3065): do_action()",
        "#20 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/includes/editor-templates/editor-wrapper.php(31): wp_head()",
        "#21 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/core/editor/loader/v1/editor-v1-loader.php(77): include('/home/wwwcustom...')",
        "#22 /home/XXX/XXX.ch/htdocs/XXX/wp-content/plugins/elementor/core/editor/editor.php(169): Elementor\Core\Editor\Loader\V1\Editor_V1_Loader->print_root_template()",
        "#23 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/class-wp-hook.php(324): Elementor\Core\Editor\Editor->init()",
        "#24 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()",
        "#25 /home/XXX/XXX.ch/htdocs/XXX/wp-includes/plugin.php(517): WP_Hook->do_action()",
        "#26 /home/XXX/XXX.ch/htdocs/XXX/wp-admin/admin.php(419): do_action()",
        "#27 /home/XXX/XXX.ch/htdocs/XXX/wp-admin/post.php(12): require_once('/home/wwwcustom...')",
        "#28 {main}",
        "thrown"
    ]
}
